### ISSUE TITLE
Rq tasks failed with `OperationalError: SSL SYSCALL error: EOF detected`

### ISSUE THREAD

**Issue by imankulov** [translate/pootle#4200]
Can you please help me to solve this. I'm trying to import data to Pootle. I set up a clean installation and use PostgreSQL as a data store. When I import data, a bunch of failed rq tasks appear. I was trying to re-run these tasks with `pootle retry_failed_jobs`, and every time ended up with a traceback looking a bit like:

```pytb
2015-11-19 17:43:48,719 INFO Loading custom settings from '/home/pootle/.pootle/pootle.conf'...
2015-11-19 17:43:49,141 INFO Using Python PO
[2015-11-19T17:43:49]	RQ worker u'rq:worker:ip-192-168-1-182.5952' started, version 0.5.6
[2015-11-19T17:43:49]
[2015-11-19T17:43:49]	*** Listening on default...
[2015-11-19T17:43:49]	default: pootle.core.mixins.treeitem.update_cache_job(<Store: /he/myproject_ios/AFNetworking.strings.po>) (5cd8a39f-01b5-4cdc-a364-777d7aee2109)
[2015-11-19T17:43:49]	Job OK
[2015-11-19T17:43:49]	Result is kept for 500 seconds
[2015-11-19T17:43:49]
[2015-11-19T17:43:49]	*** Listening on default...
[2015-11-19T17:43:49]	default: pootle.core.mixins.treeitem.update_cache_job(<Store: /he/myproject_ios/AppiraterLocalizable.strings.po>) (65adf546-d9b4-4400-86cc-d7cc8a122826)
[2015-11-19T17:43:49]	DatabaseError: SSL error: decryption failed or bad record mac

Traceback (most recent call last):
  File "/home/pootle/env/local/lib/python2.7/site-packages/rq/worker.py", line 568, in perform_job
    rv = job.perform()
  File "/home/pootle/env/local/lib/python2.7/site-packages/rq/job.py", line 495, in perform
    self._result = self.func(*self.args, **self.kwargs)
  File "/home/pootle/env/local/lib/python2.7/site-packages/pootle/core/mixins/treeitem.py", line 684, in update_cache_job
    instance._update_cache_job(keys, decrement)
  File "/home/pootle/env/local/lib/python2.7/site-packages/pootle/core/mixins/treeitem.py", line 529, in _update_cache_job
    self.update_cached(key)
  File "/home/pootle/env/local/lib/python2.7/site-packages/pootle/core/mixins/treeitem.py", line 49, in _statslog
    result = function(instance, *args, **kwargs)
  File "/home/pootle/env/local/lib/python2.7/site-packages/pootle/core/mixins/treeitem.py", line 307, in update_cached
    self.set_cached_value(name, self._calc(name))
  File "/home/pootle/env/local/lib/python2.7/site-packages/pootle/core/mixins/treeitem.py", line 209, in _calc
    return self._calc_last_updated()
  File "/home/pootle/env/local/lib/python2.7/site-packages/pootle/core/mixins/treeitem.py", line 185, in _calc_last_updated
    [self._get_last_updated()] +
  File "/home/pootle/env/local/lib/python2.7/site-packages/pootle/apps/pootle_store/models.py", line 2338, in _get_last_updated
    max_unit = self.unit_set.all().order_by('-creation_time')[0]
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/models/query.py", line 177, in __getitem__
    return list(qs)[0]
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/models/query.py", line 141, in __iter__
    self._fetch_all()
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/models/query.py", line 966, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/models/query.py", line 265, in iterator
    for row in compiler.results_iter():
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 701, in results_iter
    for rows in self.execute_sql(MULTI):
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 787, in execute_sql
    cursor.execute(sql, params)
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
DatabaseError: SSL error: decryption failed or bad record mac

Traceback (most recent call last):
  File "/home/pootle/env/local/lib/python2.7/site-packages/rq/worker.py", line 568, in perform_job
    rv = job.perform()
  File "/home/pootle/env/local/lib/python2.7/site-packages/rq/job.py", line 495, in perform
    self._result = self.func(*self.args, **self.kwargs)
  File "/home/pootle/env/local/lib/python2.7/site-packages/pootle/core/mixins/treeitem.py", line 684, in update_cache_job
    instance._update_cache_job(keys, decrement)
  File "/home/pootle/env/local/lib/python2.7/site-packages/pootle/core/mixins/treeitem.py", line 529, in _update_cache_job
    self.update_cached(key)
  File "/home/pootle/env/local/lib/python2.7/site-packages/pootle/core/mixins/treeitem.py", line 49, in _statslog
    result = function(instance, *args, **kwargs)
  File "/home/pootle/env/local/lib/python2.7/site-packages/pootle/core/mixins/treeitem.py", line 307, in update_cached
    self.set_cached_value(name, self._calc(name))
  File "/home/pootle/env/local/lib/python2.7/site-packages/pootle/core/mixins/treeitem.py", line 209, in _calc
    return self._calc_last_updated()
  File "/home/pootle/env/local/lib/python2.7/site-packages/pootle/core/mixins/treeitem.py", line 185, in _calc_last_updated
    [self._get_last_updated()] +
  File "/home/pootle/env/local/lib/python2.7/site-packages/pootle/apps/pootle_store/models.py", line 2338, in _get_last_updated
    max_unit = self.unit_set.all().order_by('-creation_time')[0]
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/models/query.py", line 177, in __getitem__
    return list(qs)[0]
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/models/query.py", line 141, in __iter__
    self._fetch_all()
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/models/query.py", line 966, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/models/query.py", line 265, in iterator
    for row in compiler.results_iter():
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 701, in results_iter
    for rows in self.execute_sql(MULTI):
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 787, in execute_sql
    cursor.execute(sql, params)
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/pootle/env/local/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
DatabaseError: SSL error: decryption failed or bad record mac

[2015-11-19T17:43:49]	Moving job to u'failed' queue
```

As you can see, while the first task succeeds, everything else fails with `DatabaseError: SSL error: decryption failed or bad record mac` or `OperationalError: SSL error: decryption failed or bad record mac`. [Here on StackOverflow](http://stackoverflow.com/q/17523912/848010) someone logs their solution: one has to reset a database connection in the beginning of the job. Not sure if it's been implemented in Pootle, or if there is a better, more elegant solution.

The output of `pip freeze` is:

```
click==5.1
cssmin==0.2.0
decorator==4.0.4
diff-match-patch==20121119
Django==1.7.10
django-allauth==0.24.1
django-assets==0.11
django-contact-form==1.0
django-overextends==0.4.0
django-redis==4.3.0
django-rq==0.8.0
django-transaction-hooks==0.2
elasticsearch==1.6.0
ipython==4.0.0
ipython-genutils==0.1.0
lxml==3.5.0
oauthlib==1.0.3
path.py==8.1.2
pexpect==4.0.1
pickleshare==0.5
Pootle==2.7.2
psycopg2==2.6.1
ptyprocess==0.5
python-openid==2.2.5
raven==5.8.1
redis==2.10.5
requests==2.8.1
requests-oauthlib==0.5.0
rq==0.5.6
simplegeneric==0.8.1
six==1.10.0
sphinx-me==0.3
traitlets==4.0.0
translate-toolkit==1.13.0
urllib3==1.12
webassets==0.11
```

Grateful for all kinds of help. Thanks!

**Comment by imankulov** [comment]
For the record. Solution [from this thread](https://github.com/ui/django-rq/issues/123#issuecomment-137647556) helps me

    pootle rqworker --worker-class rq.SimpleWorker

But I guess there should be a cleaner solution. Thanks!

**Comment by julen** [comment]
We are running on MySQL but this can be related, and you might want to give it a go: in terms of connection handling in jobs there was a recent change to avoid timeouts, merged in #4094 (it's in master right now, not part of any public releases). This wraps jobs with a check to close unusable and obsolete connections.

**Comment by imankulov** [comment]
Tested against master and can confirm that it works now (at least, in my environment). Thank you! Closing this down.
